### PR TITLE
Take node.name into consideration in autoconfiguration

### LINKED ISSUE #80465: Use node.name instead of `HOSTNAME` for auto-generated certificates
When `node.name` is set, it's most likely the this is what the user wants to identify the node by, rather than the `hostname` of the machine this node runs on. In such cases, we should set the CN of the autogenerated HTTP certifiate to `node.name`.

### PR DESCRIPTION
**Pull request by jkakavas**
If node.name is set by the user, use its value when setting
cluster.initial_master_nodes (otherwise the node will fail to start)
While doing this, also set the CN of the auto-generated TLS
certificate for the HTTP layer to the node.name value too.

Resolves: #80465, #79881

### CODE REVIEW

File: x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java [hunk 1]
```diff
@@ -635,7 +636,11 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
                             bw.newLine();
                             bw.write("# and all the subsequent nodes should be added via the node enrollment flow");
                             bw.newLine();
-                            bw.write(ClusterBootstrapService.INITIAL_MASTER_NODES_SETTING.getKey() + ": [\"${HOSTNAME}\"]");
+                            bw.write(
+                                ClusterBootstrapService.INITIAL_MASTER_NODES_SETTING.getKey()
```
Comment (albertzaharovits) [inline x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java:640]: I would adjust at least one of the packaging tests to the effect that the 'node.name' setting is randomly set, and the auto-configurations honors it (ie the initial node starts and forms a cluster).

**Review by albertzaharovits — APPROVED** [review]
LGTM

**Comment by jkakavas** [comment]
@elasticmachine test this please

**Comment by jkakavas** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-unix-sample

**Comment by jkakavas** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-windows-sample

**Comment by jkakavas** [comment]
@elasticmachine test this please

**Comment by jkakavas** [comment]
I'm purposefully running multiple invocations of the CI tests in this PR even though this is already green. Packaging tests are notoriously hard to debug/fix and I don't want to introduce another flaky one

File: x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java (hunk 1 above)
Comment (jkakavas) [reply x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java:640]: Thanks that made total sense, I added https://github.com/elastic/elasticsearch/pull/81007/files#diff-fcbff32962b19f8c6dfd6bd30c53705f4745d328bc8797382d19e1aaa18c57d4R263

**Pull request merged**